src/rules/maxLineLengthRule.ts

```diff
@@ -96,32 +108,98 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     private getRuleOptions(): MaxLineLengthRuleOptions {
-        const argument = this.ruleArguments[0];
-        let options: MaxLineLengthRuleOptions = { limit: 0 };
+        const argument = this.ruleArguments[0] as MaxLineLengthRuleOptions | number;
+
         if (typeof argument === "number") {
-            options.limit = argument;
+            return { [OPTION_LIMIT]: argument };
         } else {
-            options = argument as MaxLineLengthRuleOptions;
-            const ignorePattern = (argument as { [key: string]: string })["ignore-pattern"];
-            options.ignorePattern =
-                typeof ignorePattern === "string" ? new RegExp(ignorePattern) : undefined;
+            const {
+                [OPTION_CHECK_REGEX]: checkRegex,
+                [OPTION_CHECK_STRINGS]: checkStrings,
+                [OPTION_IGNORE_PATTERN]: ignorePattern,
+                [OPTION_LIMIT]: limit,
+            } = argument;
+
+            return {
+                [OPTION_CHECK_REGEX]: Boolean(checkRegex),
+                [OPTION_CHECK_STRINGS]: Boolean(checkStrings),
+                [OPTION_IGNORE_PATTERN]:
+                    typeof ignorePattern === "string" ? new RegExp(ignorePattern) : undefined,
+                [OPTION_LIMIT]: Number(limit),
+            };
         }
-        options.limit = Number(options.limit); // user can pass a string instead of number
-        return options;
     }
 }
 
 function walk(ctx: Lint.WalkContext<MaxLineLengthRuleOptions>) {
-    const limit = ctx.options.limit;
-    const ignorePattern = ctx.options.ignorePattern;
-    for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength <= limit) {
-            continue;
+    const { [OPTION_LIMIT]: limit } = ctx.options;
+
+    getLineRanges(ctx.sourceFile)
+        .filter(({ contentLength }: LineRange): boolean => contentLength > limit)
+        .filter((line: LineRange): boolean => !shouldIgnoreLine(line, ctx))
```

Comment:
Somewhat nitpicking here: could you please combine these two `.filter` statements? Seems odd to have two `.filter`s in a row when you could use a `&&` instead.